util/llbutil/authprovider/podman_test.go

```diff
@@ -60,6 +60,7 @@ func TestPodmanProvider(topT *testing.T) {
 	defer o.Run()
 
 	o.AfterEach(func(tt testCtx) {
+		<-tt.result
```

Comment:
I don't understand this -- this seems to be discarding the results of a test without verifying them. Other parts in this test call things like:

```
			select {
			case res := <-tt.result:
				_, ok := res.(credentials)
				tt.expect(ok).To(beTrue())
```

Reply:
`AfterEach` runs after each test in `TableSpec`. In those tests, the `result` chan is indeed read from. The new addition then waits for `result` to be closed with `<-tt.result`. The latter read is meant to verify the `close(tt.result)` occurred.

Reply:
I had to look over this test code a bit more closely; it feels more complicated than it needs to be (I'm not suggesting we change it).

I would feel a bit more comfortable with:

```
_, ok := <-tt.result
tt.expect(ok).To(beFalse()) // ensure the channel was closed
```

to prevent cases where we accidentally leave (unchecked) results on the channel.

Reply:
Yes, I agree that it's overly complicated. Sure, I'm fine with that as well.

Reply:
Updated.